filesystem.go

```diff
@@ -0,0 +1,59 @@
+package mmark
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+// FileSystem implements access to files. The elements in a file path are
+// separated by slash ('/', U+002F) characters, regardless of host
+// operating system convention.
+type FileSystem interface {

```

Comment:
It depends whether the `mmark.ParseAt` will stay as it is currently. Technically it could take a directory name as the argument and then everything in filesystem.go doesn't need to be exported. The exporting of FileSystem seemed more idiomatic code.